subsys/net/ip/net_shell.c

```diff
@@ -3659,11 +3826,60 @@ static int cmd_net_ip_add(const struct shell *shell, size_t argc, char *argv[])
 #else
 		net_if_ipv4_addr_add(iface, &addr, NET_ADDR_MANUAL, 0);
 #endif
+
+		if (argv[3]) {
+			/* netmask is provided */
+			if (net_addr_pton(AF_INET, argv[3], &addr)) {
+				PR_ERROR("Invalid netmask: %s", argv[3]);
+			} else {
+				net_if_ipv4_set_netmask(iface, &addr);
+				goto done;
+			}
+		}
+		net_addr_pton(AF_INET, CONFIG_NET_CONFIG_MY_IPV4_NETMASK, &addr);
+		net_if_ipv4_set_netmask(iface, &addr);
+done:
 		return 0;
 		}
+		PR_ERROR("Need address and netmask(optional)\n");
 		return -1;
 	}
 #endif /* CONFIG_NET_NATIVE_IPV4 */
+return 0;
+}
+
+static int cmd_net_ip_del(const struct shell *shell, size_t argc, char *argv[])
```

Comment:
IMO this belongs to the previous commit (IPv4 changes).

Reply:
Pushed it as part of IPv4 changes